Assets/Scripts/Puzzle/EarthRoom/ShieldChecker.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using RogueApeStudios.SecretsOfIgnacios.Interactables.Earth;
+using UnityEngine;
+
+namespace RogueApeStudios.SecretsOfIgnacios.Puzzle.EarthRoom
+{
+    public class ShieldChecker : MonoBehaviour
+    {
+        [SerializeField] private string _shieldTag = "Shield";
+        [SerializeField] private bool _checkForShrunk;
+        [SerializeField] private bool _checkForGrown;
+        [SerializeField] private bool _shieldFits;
+        //Each shield is different so you should add the correct shield here
+        [SerializeField] private GameObject _targetShield;
+        
+        public bool ShieldFits => _shieldFits;
+        //Action so it doesn't need to constantly verify in update
+        public event Action onShieldFitChanged;
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.CompareTag(_shieldTag) && other.TryGetComponent(out Resizable resizable) && other.gameObject == _targetShield)
```

Comment:
`[SerializeField] private Resizable _targetResizable;` instead of `TryGetComponent`, as long as the gameobject and the reference for the resizable are the same it should be fine. Also what should happen when it is not the correct shield (placeholder logs are fine for now)?